### ISSUE TITLE
Support `dd` functionality

### ISSUE THREAD

**Issue by ellcs** [mdr/mash#2]
```
Error: No binding for 'dd'. Did you mean 'cd'?
dd
^-
```

I just assume that every unix program needs it's own binding. Where do add a binding and how do i write one?

Cheers!

**Comment by mdr** [comment]
> I just assume that every unix program needs it's own binding

Yeah, the intention is that Mash should provide built-in support for most standard Unix functionality, including programs like `dd`, although it might end up with a different interface than the traditional command (`dd` actually being a good example that it has different argument convention to most programs).

> Where do add a binding and how do i write one?

At the moment, for standard lib functions, they are written in Scala. There's an example binding to `open` in `src/main/scala/com/github/mdr/mash/ns/os/OpenFunction.scala`

You could also write a `dd` binding in Mash, as a user function. A simple one might look like:

```
def dd (in="/dev/stdin") (out="/dev/stdout") (blockSize=512) (count=1) = 
  !!{dd "if=$in" "of=$out" "bs=$blockSize" "count=$count"}
```

(And if you just want to run the `dd` executable, you can also do that using the "mish" subsyntax, e.g. by prepending an `!` at the start: `!dd if=/dev/urandom of=myrandom bs=100 count=1`)
